six/modules/c++/six.sicd/tests/test_add_additional_des.cpp

```diff
@@ -0,0 +1,252 @@
+    segmentWriter->attachSource(sSource);
+    writer.addAdditionalDES(segmentWriter);
+
+    std::string name(std::tmpnam(NULL));
+    try
+    {
+        writer.save(reinterpret_cast<const six::UByte*>(bandData.get()), name);
+        std::cout << "Test passed" << std::endl;
+        std::remove(name.c_str());
+        return true;
+    }
+    catch (except::Exception& e)
+    {
+        std::cerr << e.getMessage() << std::endl;
+        std::cerr << "Test failed" << std::endl;
+        return false;
+    }
+}
+
+bool canAddTwoSegmentWriters(const std::string& xmlName)
+{
+    std::cout << "Running canAddTwoSegmentWriters\n";
+    std::auto_ptr<six::Data> data = six::sicd::Utilities::readXML(xmlName);
+    mem::ScopedArray<sys::Int16_T> bandData(
+        generateBandData(*data));
+
+    six::Container container(six::DataType::COMPLEX);
+    container.addData(data.release());
+
+    six::NITFWriteControl writer;
+    writer.initialize(&container);
+
+    nitf::Record record = writer.getRecord();
+    nitf::DESegment desOne = record.newDataExtensionSegment();
+    desOne.getSubheader().getFilePartType().set("DE");
+    desOne.getSubheader().getTypeID().set("XML_DATA_CONTENT");
+    desOne.getSubheader().getVersion().set("01");
+    desOne.getSubheader().getSecurityClass().set("U");
+
+    static const char segmentOneData[] = "123456789ABCDEF0";
+    nitf::SegmentMemorySource sOneSource(segmentOneData, strlen(segmentOneData),
+        0, 0, true);
+    mem::SharedPtr<nitf::SegmentWriter> segmentOneWriter(new nitf::SegmentWriter);
+    segmentOneWriter->attachSource(sOneSource);
+    writer.addAdditionalDES(segmentOneWriter);
+
+    nitf::DESegment desTwo = record.newDataExtensionSegment();
+    desTwo.getSubheader().getFilePartType().set("DE");
+    desTwo.getSubheader().getTypeID().set("XML_DATA_CONTENT");
+    desTwo.getSubheader().getVersion().set("01");
+    desTwo.getSubheader().getSecurityClass().set("U");
+
+    static const char segmentTwoData[] = "123456789ABCDEF0";
+    nitf::SegmentMemorySource sTwoSource(segmentTwoData, strlen(segmentTwoData),
+        0, 0, true);
+    mem::SharedPtr<nitf::SegmentWriter> segmentTwoWriter(new nitf::SegmentWriter);
+    segmentTwoWriter->attachSource(sTwoSource);
+    writer.addAdditionalDES(segmentTwoWriter);
+
+    std::string name(std::tmpnam(NULL));
+    try
+    {
+        writer.save(reinterpret_cast<const six::UByte*>(bandData.get()), name);
+        std::cout << "Test passed" << std::endl;
+        std::remove(name.c_str());
+        return true;
+    }
+    catch (except::Exception& e)
+    {
+        std::cerr << e.getMessage() << std::endl;
+        std::cerr << "Test failed" << std::endl;
+        return false;
+    }
+}
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        validateArguments(argc, argv);
+        std::string xmlName(argv[1]);
+
+        bool allPassed = true;
+        allPassed = addingNullSegmentWriterShouldThrow(xmlName) && allPassed;
```

Comment:
I think this will be a lot easier for me to add in once I’ve figured out what’s going on with the NITFReadControl. Other changes are made.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Tuesday, May 24, 2016 8:12 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Add method to allow adding extra DES to NITFWriteControl (#76)


In six/modules/c++/six.sicd/tests/test_add_additional_des.cpp<http://cp.mcafee.com/d/k-Kr410e4zqb3z2bbWdQPtPrb38UsqenTPrUVBxB4QsCzASztWXPxEVKqenzqabaarVEVsovdydorcDIel_ok1iG6JQVsSGxHtendE44foo7fZvAmkkmbCzDHTbELfIYCUejoKeVqWtAklrKcsG7DR8OJMddECSjt-hsd7bOabatTztPpesRG9px6kOrp3BYwhQzZpgDB91Vs-h2eNboISuvwVM349kITiAGM5oKcc6zBMQsLECO5mUm-w0nC7C4miXb2eBN6IdCjS7a_Ia7aEqTjBMlgy9YhBcCSgVsSUyMMr1vF6y0QJM8Wh-IEjOAwYKv8x7oBziWq81110Qga_cDl0QgjGq8dffGod46E8iwhd47xYUWq80Uq63h1eFEw1wq2hs1oODUBm4URfd41EE0MsUqajpEVKrE6_QBvj9>:

> +    {

> +        std::cerr << e.getMessage() << std::endl;

> +        std::cerr << "Test failed" << std::endl;

> +        return false;

> +    }

> +}

> +

> +int main(int argc, char** argv)

> +{

> +    try

> +    {

> +        validateArguments(argc, argv);

> +        std::string xmlName(argv[1]);

> +

> +        bool allPassed = true;

> +        allPassed = addingNullSegmentWriterShouldThrow(xmlName) && allPassed;

Can you easily add a test where you use the 5 and 283 length versions of XML_DATA_CONTENT<http://cp.mcafee.com/d/5fHCMUedEIec8ILETjdTdIIczxNEVvvdLzCm6kjhOqejqdTHLe6zCVEVudEEIEFLCzBNxYS8RxIOuMVnZxg5aEqTjBPqG6JQVsSwggZxws_R-hphhoKqeuLsKyY-POrwVdyUXBHFShhlKUNOEuvkzaT0QSCrpdTV5MQsL8EIFTudTdAVPmEBC4pj9JAenNl3mWp2kURMwvkxyKQDYslGwSvZcJA9jtOxfo-xw33ol5gUg18waygo1w96wo2uIr8lrxrW01uouohpbII8Wn4qMSpfosH-MEsGxHten1l28DN6kOrp3BPryb31I5-Aq83iT0zF7WOxfai3OVYy4tymdbFEw4443h0HYOtk3h1eFEwQY-FwQgqwxa14Qgu7PzFEw3xEod44WCy061E95M5zavylojzkYQg6yw31PxEFdCzCVKkVdEt7IKZca>? I'm not actually sure how you tell NITRO to do this, so if it's not straightforward, you can punt for now.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/2DRPowrhosohpvhKCrKrpop73zhO--rv7cIcECzAQsCQrLnusd7dPhOYrhhphjvd7bz3VIhH3pAZxOLX2walgRKDbCRkdrFOVJ0wxX30V_HYyOyyNsQsZuVt5VZDAT1Or5NTbnjIyyHtNzBgY-F6lK1FJASOrLObxEVuhhpjKYrKr9PCJhbc8OCjr8sLA2eAvHa4YF8fbDO8hS9r5CPPY7eoX8_WogKqekXLc3xRxPW2ap1DXK6zBZ_Tfhgvu7e-VtCPZQgcyUMMqen3hO-yr8lrxrW01uouohpbII8Wn4qMSpfosH-MEsGxHten1l28DN6kOrp3BPryb31I5-Aq83iT0zF7WOxfai3OVYy4tymdbFEw4443h0HYOtk3h1eFEwQY-FwQgqwxa14Qgu7PzFEw3xEod44WCy061E95M5zavylojzkYQg6yw31PxEFdCzCVIl289AkBXmOq>


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.